applications/callflow/src/module/cf_directory.erl

```diff
@@ -434,36 +445,61 @@ get_directory_user(U, CallflowId) ->
                    ,name_audio_id = kz_json:get_value(?RECORDED_NAME_KEY, U)
                    }.
 
--spec filter_users(directory_users(), kz_term:ne_binary(), 'last' | 'first') -> directory_users().
-filter_users(Users, DTMFs, 'last') ->
-    lager:info("filtering users by ~s", [DTMFs]),
-    Size = byte_size(DTMFs),
-    queue:to_list(
-      lists:foldl(fun(U, Q) -> maybe_queue_user(U, Q, DTMFs, Size, 'last') end
-                 ,queue:new()
-                 ,Users
-                 )
-     );
-filter_users(Users, DTMFs, 'first') ->
+-spec sort_users(directory_users(), 'first' | 'last') -> directory_users().
+sort_users(Users, Order) ->
+    lists:sort(fun(User1, User2) ->
```

Comment:
We typically like to break things out into named functions. At first glance I would try something like:

```erlang
sort_users(Users, 'first') ->
    lists:sort(fun sort_by_first/2, Users);
sort_users(Users, 'last') ->
    lists:sort(fun sort_by_last/2, Users).
```

Now a reader can see that the sort function used is toggled by the `Order`.

Now the sort functions themselves can pick apart the `directory_user` record as they see fit:

```erlang
sort_by_first(#directory_user{first_name=AFirst, last_name=ALast}, #directory_user{first_name=AFirst, last_name=BLast}) ->
    ALast < BLast;
sort_by_first(#directory_user{first_name=AFirst}, #directory_user{first_name=BFirst}) ->
    AFirst < BFirst.
```

Similar structure for `sort_by_last/2`. 

What do you think?